runelite-client/src/main/java/net/runelite/client/plugins/itemstats/ItemStatOverlay.java

```diff
@@ -186,33 +186,37 @@ private String getChangeString(
 		return label + ": " + ColorUtil.wrapWithColorTag(prefix + valueString + suffix, color) + "</br>";
 	}
 
-	private String buildStatBonusString(ItemStats s)
+	private String buildStatBonusString(ItemStats s, int groupId)
 	{
 		final StringBuilder b = new StringBuilder();
 		b.append(getChangeString("Weight", s.getWeight(), true, false));
 
 		ItemStats other = null;
-		final ItemEquipmentStats currentEquipment = s.getEquipment();
 
-		ItemContainer c = client.getItemContainer(InventoryID.EQUIPMENT);
-		if (s.isEquipable() && currentEquipment != null && c != null)
+		if (!(config.showInEquipmentGroup() && groupId == WidgetInfo.EQUIPMENT.getGroupId()))
 		{
-			final Item[] items = c.getItems();
-			final int slot = currentEquipment.getSlot();
+			final ItemEquipmentStats currentEquipment = s.getEquipment();
+			ItemContainer c = client.getItemContainer(InventoryID.EQUIPMENT);
 
-			if (slot != -1 && slot < items.length)
+			if (s.isEquipable() && currentEquipment != null && c != null)
```

Comment:
If I'm reading this correctly, all you need is to make `int groupId` to -> `boolean isEquipment` and then add `&& isEquipment` to here.